### PR TITLE
Fix base leaks not syncing after repair

### LINKED ISSUE #2769: Base leaks are not synced
### Which version are you playing on?

1.8.1.0 (Latest release)

### Details about version

_No response_

### Which OS are you playing on?

Windows

### On which platform did you buy Subnautica?

Steam

### Describe the issue.

Base is leaking upon server restart. Fixed from player 1. Then once player 2 joins base is still flooding for them. 

### Describe the steps you took to encounter the bug.

Started playing the day after playing for a few days. I joined and my base integrity is around 32%. I have 2 leaks. I fix the leaks. Player 2 then joined. The base was leaking for them. They then repaired the same leaks and the base is fixed for them. Now once we receive a hole in the base we both have to fix that exact same spot. Have tried the base resync. Once we both log off and join back again the base is still leaking. Same 2 holes. Any help? 

### Relevant log output

[server\[PooleFam\]-20260529.log](https://github.com/user-attachments/files/28414821/server.PooleFam.-20260529.log)

[game\[Jonathan\]-20260529.log](https://github.com/user-attachments/files/28414823/game.Jonathan.-20260529.log)

### PR DESCRIPTION
**Pull request by misterbubb**
Closes #2769

## Description
Fixes base leaks not syncing properly between players and persisting after server restarts.

## Problem
When Player 1 repaired base leaks, Player 2 would still see the same leaks when joining. Additionally, after server restart, the repaired leaks would reappear for all players. This was caused by leak entities being destroyed from memory but not persisted to disk until the next autosave cycle.

## Solution
**Server-side:**
- Added debounced save trigger in `LeakRepairedProcessor` that queues a save within 5 seconds of leak repair
- Ensures leak deletions are persisted immediately, preventing reappearance after crashes or player joins
- Debouncing prevents excessive saves when multiple leaks are repaired rapidly

**Client-side:**
- Added defensive health check in `BaseLeakEntitySpawner` to skip spawning fully repaired leaks (Health >= 100)
- Provides additional safety layer for edge cases where save hasn't persisted yet

## Testing
Test 1: Player 1 repairs leaks → Player 2 joins → No leaks appear
Test 2: Player 1 repairs leaks → Server restart → No leaks reappear

### CODE REVIEW

File: Nitrox.Server.Subnautica/Models/Packets/Processors/LeakRepairedProcessor.cs [hunk 1]
```diff
@@ -1,17 +1,50 @@
...
+                lastSaveRequest = now;
+                shouldTriggerSave = true;
+            }
+        }
+
+        if (shouldTriggerSave)
+        {
+            logger.ZLogDebug($"Queueing save after base leak repair to ensure persistence");
+            await saveService.QueueActionAsync(SaveService.ServiceAction.SAVE);
         }
```
Comment (Measurity) [inline Nitrox.Server.Subnautica/Models/Packets/Processors/LeakRepairedProcessor.cs:48]: Leak repair shouldn't trigger a save, it's not related to world saving. 

Data loss on crash isn't preventable and users can already change the save interval to reduce data loss.

**Review by Measurity — CHANGES_REQUESTED** [review]
> Provides additional safety layer for edge cases where save hasn't persisted yet

This shouldn't happen. Server uses latest data from memory on joins. Save files are only used when the server starts up. If not, we have a design problem (race condition or otherwise) and should fix this instead.

File: Nitrox.Server.Subnautica/Models/Packets/Processors/LeakRepairedProcessor.cs (hunk 1 above)
Comment (misterbubb) [reply Nitrox.Server.Subnautica/Models/Packets/Processors/LeakRepairedProcessor.cs:48]: Wait yeah your right, I confirmed the entity removal is working correctly. closing this PR now, sorry!

File: NitroxClient/GameLogic/Spawning/Bases/BaseLeakEntitySpawner.cs [hunk 2]
```diff
@@ -32,6 +32,14 @@ protected override bool SpawnSync(BaseLeakEntity entity, TaskResult<Optional<Gam
             return true;
         }
 
+        // Skip spawning leaks that are already at full health (fully repaired)
+        // This handles edge cases where a leak was repaired but the save hasn't persisted yet
+        if (entity.Health >= 100f)
+        {
+            Log.Debug($"[{nameof(BaseLeakEntitySpawner)}] Skipping spawn of fully repaired leak {entity.Id} (Health: {entity.Health})");
+            return true;
```
Comment (Measurity) [inline NitroxClient/GameLogic/Spawning/Bases/BaseLeakEntitySpawner.cs:40]: This might still be worth having though apart from other changes.

Comment (jonathandwaynepoole796-art) [reply NitroxClient/GameLogic/Spawning/Bases/BaseLeakEntitySpawner.cs:40]: Can you walk me through what i need to do to get this fixed?

Comment (dartasen) [reply NitroxClient/GameLogic/Spawning/Bases/BaseLeakEntitySpawner.cs:40]: @jonathandwaynepoole796-art I think the first part is to understand what's happening, by testing and looking at the data stored server-side (either by debugging it or looking at the json files)

* When a leak is created, which entity is created, are the informations correct ? (health, position, rotation, ...)
* When the same leak is repaired, what happen to the data (is the entity deleted, have weird data, ...) ?
* When a client repairs the leak, do we emit a packet that will be processed correctly by the server
* ...

With those informations it's easier to create a chirurgical fix

Maybe @misterbubb have more informations
